addons/purchase_stock/tests/test_stockvaluation.py

```diff
@@ -764,6 +764,71 @@ def test_standard_valuation_multicurrency(self):
         self.assertAlmostEqual(picking_aml.amount_currency, -20, msg="credit value for stock should be equal to the standard price of the product.")
         self.assertAlmostEqual(diff_aml.amount_currency, -80, msg="credit value for price difference")
 
+    def test_bill_on_ordered_qty_valuation_multicurrency(self):
+        """
+            Product with an invoice policy on ordered quantity should keep the valuation
+            of the bill with the exchange rate at the bill date
+        """
+        company = self.env.user.company_id
+        company.anglo_saxon_accounting = True
+        company.currency_id = self.usd_currency
+
+        date_po = '1993-07-18'
+
+        self.product1.product_tmpl_id.categ_id.property_cost_method = 'average'
+        self.product1.product_tmpl_id.categ_id.property_valuation = 'real_time'
+        self.product1.purchase_method = 'purchase'
+
+        self.env['res.currency.rate'].create([{
+            'name': date_po,
+            'rate': 1 / 1.5,
+            'currency_id': self.eur_currency.id,
+            'company_id': company.id,
+        }])
+
+        # Create PO
+        po = self.env['purchase.order'].create({
+            'currency_id': self.eur_currency.id,
+            'partner_id': self.partner_id.id,
+            'order_line': [
+                (0, 0, {
+                    'name': self.product1.name,
+                    'product_id': self.product1.id,
+                    'product_qty': 1.0,
+                    'product_uom': self.product1.uom_po_id.id,
+                    'price_unit': 100.0,  # 100€ = 150$
+                    'date_planned': date_po,
+                }),
+            ],
+        })
+        po.button_confirm()
+
+        # Create and post the vendor bill before recieving the product
+        self.env['account.move'].with_context(default_move_type='in_invoice').create({
+            'move_type': 'in_invoice',
+            'invoice_date': date_po,
+            'date': date_po,
+            'currency_id': self.usd_currency.id,
+            'partner_id': self.partner_id.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Test',
+                'price_unit': 100.0,  # 100$
+                'product_id': self.product1.id,
+                'purchase_line_id': po.order_line.id,
+                'quantity': 1.0,
+                'account_id': self.stock_input_account.id,
+            })]
+        }).action_post()
+
+        receipt = po.picking_ids
+        receipt.move_line_ids.qty_done = 1
```

Comment:
@HANNICHE-Walid `qty_done` is not available in v17

Reply:
It is being fixed ;)